crates/services/gas_price_service/src/v1/da_source_adapter/service.rs

```diff
@@ -30,27 +39,25 @@ where
 {
     poll_interval: Interval,
     source: Source,
-    sender: Sender<DaBlockCosts>,
-    cache: HashSet<DaBlockCosts>,
+    shared_state: DaBlockCostsSharedState,
 }
 
+const DA_BLOCK_COSTS_CHANNEL_SIZE: usize = 10;
+const POLLING_INTERVAL_MS: u64 = 10_000;
+
 impl<Source> DaBlockCostsService<Source>
 where
     Source: DaBlockCostsSource,
 {
-    pub fn new(
-        source: Source,
-        sender: Sender<DaBlockCosts>,
-        poll_interval: Option<Duration>,
-    ) -> Self {
+    pub fn new(source: Source, poll_interval: Option<Duration>) -> Self {
```

Comment:
Why `poll_interval` can be `Option`?

Reply:
we want to specify a sensible default to poll the block committer